bundles/framework/timeseries/service/TimeseriesLayerService.js

```diff
@@ -0,0 +1,146 @@
+ * Registers clazzes to be used as animators for specific layer types in timeseries
+ */
+Oskari.clazz.define(
+    'Oskari.mapframework.bundle.timeseries.TimeseriesLayerService',
+    /**
+     * @method create called automatically on construction
+     * @static
+     *
+     * @param {Oskari.Sandbox} sandbox
+     *          reference to application sandbox
+     */
+    function (sandbox, timeseriesService) {
+        var me = this,
+            p;
+
+        me._layerTypeAnimators = {};
+        me._timeseriesService = timeseriesService;
+        me._sandbox = sandbox;
+        me._popupService = sandbox.getService('Oskari.userinterface.component.PopupService');
+        me.loc = Oskari.getMsg.bind(null, 'timeseries');
+        me._log = Oskari.log('TimeseriesLayerService');
+        for (p in me.__eventHandlers) {
+            if (me.__eventHandlers.hasOwnProperty(p)) {
+                me._sandbox.registerForEventByName(me, p);
+            }
+        }
+    }, {
+        /** @static @property __qname fully qualified name for service */
+        __qname: 'Oskari.mapframework.bundle.timeseries.TimeseriesLayerService',
+        /**
+         * @method getQName
+         * @return {String} fully qualified name for service
+         */
+        getQName: function () {
+            return this.__qname;
+        },
+        /** @static @property __name service name */
+        __name: "TimeseriesLayerService",
+        /**
+         * @method getName
+         * @return {String} service name
+         */
+        getName: function () {
+            return this.__name;
+        },
+        /**
+         * @static @property {Object} __eventHandlers
+         */
+        __eventHandlers: {
+            'AfterRearrangeSelectedMapLayerEvent': function (event) {
+                this.updateTimeseriesLayers();
+            },
+            'AfterMapLayerAddEvent': function (event) {
+                this.updateTimeseriesLayers();
+            },
+            'AfterMapLayerRemoveEvent': function (event) {
+                var series = this._timeseriesService.unregisterTimeseries(event.getMapLayer().getId(), 'layer');
+                if (series) {
+                    series.delegate.destroy();
+                    this.updateTimeseriesLayers();
+                }
+            }
+        },
+        /**
+         * @method _checkMultipleLayers
+         * @private
+         * Show popup if there are more than one selected timeseries layers
+         */
+        _checkMultipleLayers: function () {
+            if (this._timeseriesService.getCountByType('layer') > 1) {
+                this._popupService.closeAllPopups(false);
+                var popup = this._popupService.createPopup();
+                var closeBtn = Oskari.clazz.create('Oskari.userinterface.component.Button');
+                closeBtn.setTitle(this.loc('alert.ok'));
+                closeBtn.setHandler(function () {
+                    popup.close(true);
+                });
```

Comment:
You can create a close button for a popup with just:

       var closeBtn = popup.createCloseButton(this.loc('alert.ok'));

The label parameter is optional defaulting to a "Close" label.